File: mobile/src/transport/ScaleTransport.ts
import type { ConnectionStateValue } from '../constants/ble';

export type ScaleTransportListener = {
  onConnectionStateChanged: (state: ConnectionStateValue) => void;
  onWeightData: (weight: number) => void;
};

export interface ScaleTransport {
  setListener(listener: ScaleTransportListener | null): void;
  sendTare(): void;
  sendCalibrate(refMassGrams: number): void;
  close(): void;
}

File: android-legacy/app/src/main/java/dev/talk2scale/ScaleTransport.java
package dev.talk2scale;

public interface ScaleTransport {

    interface Listener {
        void onConnectionStateChanged(int state);
        void onWeightData(int weight);
    }

    void setListener(Listener listener);

    void sendTare();

    void sendCalibrate(int refMassGrams);

    void close();
}
